File: src/dynamic-programming/arithmetic-slices.ts
export function numberOfArithmeticSlices(nums: number[]): number {
  if (nums.length < 3) return 0
  let t = 0
  let ans = 0

  for (let i = 2; i < nums.length; ++i) {
    if (nums[i] - nums[i - 1] === nums[i - 1] - nums[i - 2]) {
      ans += ++t
    } else {
      t = 0
    }
  }

  return ans
}

File: src/dynamic-programming/arithmetic-slices.cpp
class Solution
{
public:
    int numberOfArithmeticSlices(vector<int> &nums)
    {
        int n = nums.size();
        if (n < 3)
            return 0;
        int ans = 0;
        int t = 0;

        for (int i = 2; i < n; ++i)
        {
            if (nums[i] - nums[i - 1] == nums[i - 1] - nums[i - 2])
            {
                ans += ++t;
            }
            else
            {
                t = 0;
            }
        }

        return ans;
    }
};